src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -289,8 +306,22 @@ private static void addBody(HttpURLConnection connection, Request<?> request, by
             connection.setRequestProperty(
                     HttpHeaderParser.HEADER_CONTENT_TYPE, request.getBodyContentType());
         }
-        DataOutputStream out = new DataOutputStream(connection.getOutputStream());
+        OutputStream out = this.createOutputStream ( request, connection, body.length );
```

Comment:
This should be:

DataOutputStream out = new DataOutputStream(createOutputStream(request, connection, body.length));
out.write(body);
out.close();